### PR TITLE
[logrotate]: Relax APPL_DB check tolerance

### LINKED ISSUE #12612: [Logrotate] test case test_orchagent_logrotate could be affected by the config reload in the teardown of the previous test
**Description**

**Steps to reproduce the issue:**
1. Run the test tests/syslog/test_logrotate.py
2.
3.

**Describe the results you received:**
The first run of test test_orchagent_logrotate failed:
```
Failed: Found pending entries in APPL_DB: ['_NEIGH_TABLE:Ethernet64:10.20.30.40']
orch_logrotate_setup = (['_ROUTE_TABLE:192.194.184.128/25', '_ROUTE_TABLE:193.58.224.128/25', '_ROUTE_TABLE:192.171.132.0/25', '_ROUTE_TABLE:193.34.108.128/25', '_ROUTE_TABLE:192.240.148.128/25', '_ROUTE_TABLE:192.180.138.0/25', ...], 'Ethernet64')
rand_selected_dut = <MultiAsicSonicHost r-leopard-79>

    @pytest.mark.repeat(5)
    def test_orchagent_logrotate(orch_logrotate_setup, rand_selected_dut):
        """
        Tests for the issue where an orchagent logrotate can cause a missed APPL_DB notification
        """
        ignore_entries, target_port = orch_logrotate_setup
        rand_selected_dut.control_process('orchagent', pause=True)
        rand_selected_dut.control_process('orchagent', signal='SIGHUP')
        rand_selected_dut.shell('sudo ip neigh add {} lladdr {} dev {}'.format(FAKE_IP, FAKE_MAC, target_port))
        rand_selected_dut.control_process('orchagent', pause=False)
        pending_entries = get_pending_entries(rand_selected_dut, ignore_list=ignore_entries)
>       pytest_assert(not pending_entries, "Found pending entries in APPL_DB: {}".format(pending_entries))
E       Failed: Found pending entries in APPL_DB: ['_NEIGH_TABLE:Ethernet64:10.20.30.40']

ignore_entries = ['_ROUTE_TABLE:192.194.184.128/25', '_ROUTE_TABLE:193.58.224.128/25', '_ROUTE_TABLE:192.171.132.0/25', '_ROUTE_TABLE:193.34.108.128/25', '_ROUTE_TABLE:192.240.148.128/25', '_ROUTE_TABLE:192.180.138.0/25', ...]
orch_logrotate_setup = (['_ROUTE_TABLE:192.194.184.128/25', '_ROUTE_TABLE:193.58.224.128/25', '_ROUTE_TABLE:192.171.132.0/25', '_ROUTE_TABLE:193.34.108.128/25', '_ROUTE_TABLE:192.240.148.128/25', '_ROUTE_TABLE:192.180.138.0/25', ...], 'Ethernet64')
pending_entries = ['_NEIGH_TABLE:Ethernet64:10.20.30.40']
rand_selected_dut = <MultiAsicSonicHost r-leopard-79>
target_port = 'Ethernet64'

syslog/test_logrotate.py:285: Failed
```
The root cause is that there is a config reload in the teardown of the previous test case test_logrotate_small_size:
https://github.com/sonic-net/sonic-mgmt/blob/92ea888a8fa3dc6a80f2867d3b79e587a8b6e667/tests/syslog/test_logrotate.py#L64
When the test test_orchagent_logrotate starts, the system is still processing thousands of routes, which make it impossible for the neighbor update to be processed as these are sequential.
The test should start only after the system is in a stable state.

**Describe the results you expected:**
Test should pass.

**Additional information you deem important:**

    **Output of `show version`:**

    ```
    (paste your output here)
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

### PR DESCRIPTION
**Pull request by theasianpianist**
### Description of PR

Summary:
Fixes #12612 

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311

#### What is the motivation for this PR?
The orchagent log rotate notification test can fail if orchagent is busy executing many operations (such as during system init after a config reload). This leads to a delay in processing APPL_DB notifications. If the test case checks APPL_DB during the delay, the test will fail even if orchagent later does process the notification correctly.

#### How did you do it?
Add 5 seconds of tolerance to the test case

#### How did you verify/test it?
Run the entire test module. The test case which runs before `test_orchagent_logrotate` calls `config_reload` during teardown. Prior to the fix, this would result in `test_orchagent_logrotate` failing. After the fix, all tests pass consistently.

### CODE REVIEW

File: tests/syslog/test_logrotate.py [hunk 1]
```diff
@@ -40,8 +41,8 @@ def backup_syslog(rand_selected_dut):
     logger.info('Recover syslog file to syslog')
     duthost.shell('sudo mv /var/log/syslog_bk /var/log/syslog')
 
-    logger.info('Remove temp file /var/log/syslog.1')
-    duthost.shell('sudo rm -f /var/log/syslog.1')
+    # logger.info('Remove temp file /var/log/syslog.1')
```
Comment (prsunny) [inline tests/syslog/test_logrotate.py:44]: Why is this commented? If not used, can you remove the dead code?

**Review by prsunny — COMMENTED** [review]
please fix conflict. changes lgtm

**Review by prsunny — APPROVED** [review]

**Comment by congh-nvidia** [comment]
@wangxin could you please cherry-pick this to 202405?

**Comment by echuawu** [comment]
We also are facing the failure of test_orchagent_logrotate[xxx-None-1-5], even we have this PR. Add the wait_until of no_pending_entries could not fix that. The hung and recovered orchagent process would not handle the pending entries. The root cause of the failure was after the config reload in fixture simulate_small_var_log_partition, the switch does not fully recovered, here I created a PR https://github.com/sonic-net/sonic-mgmt/pull/14478 to handle it and the fix was validated PASS at our setup.

**Pull request merged**